community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/RWLock.java

```diff
@@ -146,42 +172,53 @@ synchronized void acquireReadLock( Object tx ) throws DeadlockDetectedException
     {
         TxLockElement tle = getOrCreateLockElement( tx );
 
+        LockRequest lockRequest = null;
+        boolean addLockRequest = true;
```

Comment:
Answering second part of a question: in case interrupt was raised - means that we interrupted by somebody releasing  corresponding lock in the same time `waitingThreadList` does not contains any request from us for specified resource and we need to add it again into a queue if we will stay waiting in a while loop